plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java

```diff
@@ -266,7 +268,7 @@ protected String requestUniqueId(HttpExchange exchange) {
             }
 
             final String range = exchange.getRequestHeaders().getFirst("Content-Range");
-            return exchange.getRemoteAddress().toString()
+            return exchange.getRemoteAddress().getHostString()
```

Comment:
There seems to be a behavior change with retries here. The retries are now executed across multiple different TCP connections instead of reusing the same connection. This means we can't key them by the full address including the port any longer because that would allow for more retries than we assume.